Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/ramsey_freq_duration.py

```diff
@@ -0,0 +1,115 @@
+"""
+ramsey_freq_duration.py: template for acquiring the 2D (idle time & pulse frequency sweeps) Ramsey oscillations
+"""
+from qm.qua import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm import SimulationConfig, LoopbackInterface
+from configuration import *
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import signal
+
+##############################
+# Program-specific variables #
+##############################
+n_avg = 1000  # Number of averaging loops
+
+cooldown_time = 5 * qubit_T1 // 4  # Resonator cooldown time in clock cycles (4ns)
+
+# Frequency sweep in Hz (Needs to be a list of int)
+freq_span = 1e6
+n_freq = 41
+freq_array = (np.linspace(-freq_span / 2, freq_span / 2, n_freq) + qubit_IF).astype(int)
+
+# Idle time sweep (Needs to be a list of int)
+delay_max = 4e-6 / 1e-9
+n_delay = 101
+delay_array = np.round(np.linspace(0, delay_max, n_delay) / 4).astype(int)
+
+###################
+# The QUA program #
+###################
+with program() as ramsey_freq_duration:
+    n = declare(int)  # Averaging index
+    f = declare(int)  # Resonator frequency
+    delay = declare(int)  # Idle time
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_each_(delay, delay_array.tolist()):
+            with for_each_(f, freq_array.tolist()):
+                # Update the resonator frequency
+                update_frequency("qubit", f)
+                # Adjust the idle time
+                with if_(delay >= 4):
```

Comment:
you don't allow delays >0 <4, this is a bit confusing... Maybe add an asseration?

Reply:
Yes, this was because the wait statement needs a delay >=4 so if the delay vector contains values smaller than 4, then it will be like having zero delay just so that we were not bothered with error messages

Reply:
But this would lead to the data being "corrupted", a delay of 2 wouldn't give the correct result...
I don't like it...

Reply:
Ok, I added a test on the values of delay